components/inputs/input-text.js

```diff
@@ -618,5 +632,19 @@ class InputText extends FocusMixin(LabelledMixin(FormElementMixin(SkeletonMixin(
 
 	}
 
+	_validateUnit() {
+		clearTimeout(this._validatingUnitTimeout);
+		// don't error immediately in case it doesn't get set immediately
+		this._validatingUnitTimeout = setTimeout(() => {
```

Comment:
This turned out to be more complex than I imagined... sorry! 😬 

It looks like `_missingUnitLabelErrorHasBeenThrown` never gets reset once it's been set to `true`. Maybe we want to return early from this method if it's `true`?

```javascript
if (this._missingUnitLabelErrorHasBeenThrown) return;
this._validatingUnitTimeout = setTimeout(() => {
    ...
}, 3000);
```

Not a big deal though, since `_validateUnit()` only gets called when `unit` or `unitLabel` changes, and we probably never expect it to get called again once those properties are set.

Reply:
Yeah, this kind of validation is tricky because the calls to `updated` happen in the order that attributes are set.

I can bail early -- that would prevent checking in future calls, but it doesn't save much during component initialization since we still need to check `_missingUnitLabelErrorHasBeenThrown` inside the callback.

A typical scenario will be:
- `updated` gets called with the one attribute, we validate
- `updated` gets called with the other, we validate
- `firstUpdated` gets called, we validate
- 3 seconds elapses, so the first callback runs, then the second, then the third

Reply:
Wouldn't the first and second callbacks get cancelled due tot the `clearTimeout`?

Reply:
Hmm, yep you're right -- moved the check up higher as you suggested.